File: Java/src/main/java/com/example/sort/BubbleSort.java
package com.example.sort;

public class BubbleSort {

    static public void sort(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    SortUtil.swap(arr, j, j + 1);
                }
            }
        }
    }

    static public void sort1(int arr[]) {
        int n = arr.length;
        boolean flag = true;
        for (int i = 0; i < n - 1 && flag; i++) {
            flag = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] < arr[j + 1]) {
                    SortUtil.swap(arr, j, j + 1);
                    flag = true;
                }
            }
        }
    }
}

File: typescript/sort/bubbleSort.ts
import { compareAsc, swap } from "./sortUtils"


export function bubbleSort(arr: number[]): number[] {
    const length = arr.length
    for (let i = 0; i < length; i++) {
        for (let j = 0; j < length - i - 1; j++) {
            if (compareAsc(arr[j], arr[j + 1])) {
                swap(arr, j, j + 1)
            }
        }
    }
    return arr
}

export function bubbleSort2(arr: number[]): number[] {
    const length = arr.length
    let flag = true
    for (let i = 0; i < length && flag; i++) {
        flag = false
        for (let j = length - 1; j > i; j--) {
            if (compareAsc(arr[j], arr[j - 1])) {
                swap(arr, j, j - 1)
                flag = true
            }
        }
    }
    return arr
}